### PR TITLE
DBZ-2697 Optimisation on SQLServer connector for skipped.operations

### PR DESCRIPTION
**Pull request by blcksrx**
https://issues.redhat.com/browse/DBZ-2697

### CODE REVIEW

**Comment by blcksrx** [comment]
I ran tests locally, it was okay(always local is okay:D), but I think the test failure is irrelevant to this PR

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java [hunk 1]
```diff
@@ -65,7 +66,7 @@
     private static final String SQL_SERVER_VERSION = "SELECT @@VERSION AS 'SQL Server Version'";
     private final String lsnToTimestamp;
     private static final String INCREMENT_LSN = "SELECT sys.fn_cdc_increment_lsn(?)";
-    private static final String GET_ALL_CHANGES_FOR_TABLE = "SELECT * FROM cdc.[fn_cdc_get_all_changes_#](?, ?, N'all update old') order by [__$start_lsn] ASC, [__$seqval] ASC, [__$operation] ASC";
+    private static String GET_ALL_CHANGES_FOR_TABLE = "SELECT * FROM cdc.[fn_cdc_get_all_changes_#](?, ?, N'all update old') order by [__$start_lsn] ASC, [__$seqval] ASC, [__$operation] ASC";
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java:69]: Why isn't this final any longer? We can set final variables in the constructor, they shouldn't be static then any longer though.

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java [hunk 2]
```diff
@@ -125,6 +143,28 @@ public SqlServerConnection(Configuration config, Clock clock, SourceTimestampMod
...
+                    case UPDATE:
+                        skippedOps.add("3");
+                        skippedOps.add("4");
+                        break;
+                    case DELETE:
+                        skippedOps.add("1");
+                        break;
+                }
+            });
+            GET_ALL_CHANGES_FOR_TABLE = GET_ALL_CHANGES_FOR_TABLE.replace("__operations__", String.join(",", skippedOps));
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java:166]: While it works, this code looks unnecessarily complex to me. How about this:

```
StringBuilder getAllChangesForTableStatement = new StringBuilder("SELECT * FROM cdc.[fn_cdc_get_all_changes_#](?, ?, N'all update old') WHERE __$operation NOT IN (");

switch (operation) {
                    case CREATE:
                        getAllChangesForTableStatement.append("2");
                        break;
                    case UPDATE:
                        getAllChangesForTableStatement.append("3, 4");
                        break;
                    case DELETE:
                        getAllChangesForTableStatement.append("1");
                        break;
                }

getAllChangesForTableStatement.append(") order by [__$start_lsn] ASC, [__$seqval] ASC, [__$operation] ASC");

this.getAllChangesForTableStatement = getAllChangesForTableStatement.build();

**Review by gunnarmorling — COMMENTED** [review]
One comment inline; and do we have a test for this functionality for SQL Server? Ideally, this test continues to pass as it passed bfore, with this change in place. Thx!

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
Some comments inline. Thx, @blcksrx!

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java (hunk 1 above)
Comment (blcksrx) [reply debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java:69]: yeah, you are right, my bad

**Comment by blcksrx** [comment]
haha, everytime that I cant run test on my local, it passes here :)))

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java [hunk 3]
```diff
@@ -65,7 +66,7 @@
     private static final String SQL_SERVER_VERSION = "SELECT @@VERSION AS 'SQL Server Version'";
     private final String lsnToTimestamp;
     private static final String INCREMENT_LSN = "SELECT sys.fn_cdc_increment_lsn(?)";
-    private static final String GET_ALL_CHANGES_FOR_TABLE = "SELECT * FROM cdc.[fn_cdc_get_all_changes_#](?, ?, N'all update old') order by [__$start_lsn] ASC, [__$seqval] ASC, [__$operation] ASC";
+    private final String GET_ALL_CHANGES_FOR_TABLE;
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java:69]: The name shouldn't be upper-cased any longer, given this isn't final static any longer.

**Review by gunnarmorling — COMMENTED** [review]
LGTM overall, @blcksrx! One last tiny nit inline.

**Comment by gunnarmorling** [comment]
This also would need a rebase, @blcksrx!

**Comment by gunnarmorling** [comment]
@blcksrx, could you run the formatter, please? Always a good idea to do so locally before committing :)

**Comment by blcksrx** [comment]
@gunnarmorling My bad, sorry :)

**Comment by gunnarmorling** [comment]
There are quite a few test failures :( Do you think they are related? Does the test suite pass for you when building this change locally?

**Comment by blcksrx** [comment]
the tests were unrelated, let me rebase with master and run actions again

**Comment by gunnarmorling** [comment]
@blcksrx, hum, I'm seeing tons of test failures when building this branch :( Does the build pass for you locally?

**Comment by blcksrx** [comment]
Thanks for checking it, Im going to look at it at the weekend

**Comment by gunnarmorling** [comment]
Hey @blcksrx, hum, so there are still quite a lot of test failures. I'm going to close this one for the time being, so to clean off the slate a bit. Please feel free to re-open once all tests (at least in the SQL Server connector module) pass for you locally. Thx!

**Pull request closed without merging**